source/llassetgen-rendering/main.cpp

```diff
@@ -0,0 +1,383 @@
+    }
+
+    virtual void paintGL() override {
+        glClearColor(m_backgroundColor.r, m_backgroundColor.g, m_backgroundColor.b, m_backgroundColor.a);
+        glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+
+        glEnable(GL_BLEND);
+        glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+
+        if (m_texture) {
+            m_texture->bindActive(m_samplerIndex);
+        }
+
+        m_program->use();
+        m_program->setUniform("fontColor", m_fontColor);
+        m_vao->drawArrays(GL_TRIANGLE_STRIP, 0, 4);
+        m_program->release();
+
+        if (m_texture) {
+            m_texture->unbindActive(m_samplerIndex);
+        }
+
+        glDisable(GL_BLEND);
+    }
+
+    virtual void keyPressEvent(QKeyEvent *event) override {
+        makeCurrent();
+
+        switch (event->key()) {
+            case Qt::Key_F5:
+                m_vertexShaderSource->reload();
+                m_fragmentShaderSource->reload();
+                updateGL();
+                break;
+            case Qt::Key_Escape:
+                qApp->quit();
+                break;
+            default:
+                break;
+        }
+        doneCurrent();
+    }
+
+   public slots:
+    virtual void backgroundColorRChanged(QString value) override {
+        int red = value.toInt();
+        m_backgroundColor.r = red / 255.f;
+        paint();
+    }
+
+    virtual void backgroundColorGChanged(QString value) override {
+        int green = value.toInt();
+        m_backgroundColor.g = green / 255.f;
+        paint();
+    }
+
+    virtual void backgroundColorBChanged(QString value) override {
+        int blue = value.toInt();
+        m_backgroundColor.b = blue / 255.f;
+        paint();
+    }
+
+    virtual void fontColorRChanged(QString value) override {
+        int red = value.toInt();
+        m_fontColor.r = red / 255.f;
+        paint();
+    }
+
+    virtual void fontColorGChanged(QString value) override {
+        int green = value.toInt();
+        m_fontColor.g = green / 255.f;
+        paint();
+    }
+
+    virtual void fontColorBChanged(QString value) override {
+        int blue = value.toInt();
+        m_fontColor.b = blue / 255.f;
+        paint();
+    }
+
+   protected:
+    std::unique_ptr<globjects::Buffer> m_cornerBuffer;
+    std::unique_ptr<globjects::Program> m_program;
+    std::unique_ptr<globjects::File> m_vertexShaderSource;
+    std::unique_ptr<globjects::AbstractStringSource> m_vertexShaderTemplate;
+    std::unique_ptr<globjects::Shader> m_vertexShader;
+    std::unique_ptr<globjects::File> m_fragmentShaderSource;
+    std::unique_ptr<globjects::AbstractStringSource> m_fragmentShaderTemplate;
+    std::unique_ptr<globjects::Shader> m_fragmentShader;
+    std::unique_ptr<globjects::VertexArray> m_vao;
+
+    std::unique_ptr<globjects::Texture> m_texture;
+
+    glm::vec4 m_backgroundColor;
+    glm::vec4 m_fontColor;
+    int m_samplerIndex;
```

Comment:
Is the `m_` prefix there to stay consistent with Qt or any other library? Otherwise I would use the same style we use everywhere else.

Reply:
m_ stands for member variable, it's a coding style we use for years at the CGS, that's why I automatically used it, sorry :D
So, as far as I see, the style in this project is to simply remove the m_?